unit-tests/src/test/java/org/eclipse/collections/impl/bag/strategy/mutable/HashBagWithHashingStrategyTest.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2016 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.bag.strategy.mutable;
+
+import org.eclipse.collections.api.bag.MutableBag;
+import org.eclipse.collections.api.block.HashingStrategy;
+import org.eclipse.collections.api.tuple.primitive.ObjectIntPair;
+import org.eclipse.collections.impl.bag.mutable.HashBag;
+import org.eclipse.collections.impl.bag.mutable.MutableBagTestCase;
+import org.eclipse.collections.impl.block.factory.HashingStrategies;
+import org.eclipse.collections.impl.factory.Bags;
+import org.eclipse.collections.impl.factory.Lists;
+import org.eclipse.collections.impl.test.Verify;
+import org.eclipse.collections.impl.test.domain.Person;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * JUnit test suite for {@link HashBagWithHashingStrategy}.
+ */
+public class HashBagWithHashingStrategyTest extends MutableBagTestCase
+{
+    private static final HashingStrategy<String> STRING_HASHING_STRATEGY = HashingStrategies.nullSafeHashingStrategy(new HashingStrategy<String>()
```

Comment:
Both constant hashing strategies are the same as each other, and the same as `HashingStrategies.defaultStrategy()`. The other strategies for first name and last name allow for easy collisions, which is good for testing.